octodns/source/axfr.py

```diff
@@ -229,14 +229,16 @@ def __init__(self, id, directory, file_extension=None, check_origin=True):
         self._zone_records = {}
 
     def _load_zone_file(self, zone_name):
+
+        zone_filename = zone_name
+        if self.file_extension:
+            zone_filename = '{}{}'.format(zone_name,
+                                          self.file_extension.lstrip('.'))
+
         zonefiles = listdir(self.directory)
```

Comment:
Why get the listing of all files?  What if there are thousands of files in this directory?  This operation would be slow.  My suggestion is to check if the file exists  `if isfile(zone_filename):`